main.py

```diff
@@ -139,43 +162,45 @@ def execute():
-                purchase_button.click()
+                purchase_steps(browser)
 
-                # wait until its visible and then click the purchase button
-                logger.debug('find and click on the last purchase button')
-                WebDriverWait(browser, TIMEOUT).until(
-                    EC.visibility_of_element_located((By.XPATH, "//button[contains(@class,'btn-primary')]"))
-                ).click()
-
-                # wait until its visible and then click the 'I Agree" popup
-                # 'Refund and Right of Withdrawal Information' popup
-                try:
-                    logger.debug('accept the conditions of refund popup')
-                    WebDriverWait(browser, TIMEOUT).until(
-                        EC.visibility_of_all_elements_located((By.XPATH, "//button[contains(@class,'btn-primary')]"))
-                    )[1].click()
-                except (NoSuchElementException, LookupError) as ex:
-                    logger.debug('no refund conditions popup to accept')
-
-                # Purchase should be complete. Checking for confirmation
-                logger.debug('Wait for confirmation that checkout is complete')
-                WebDriverWait(browser, TIMEOUT).until(EC.visibility_of_element_located((
-                    By.XPATH, "//h1/span[contains(text(),'Install')]|//span[contains(text(),'Thank you for buying')]"
-                )))
-                logger.info('obtained game %s. Price was %s and %s', name, price, expires)
+                logger.info('obtained %s. Price was %s and %s', name, price, expires)
             elif purchase_button.text == 'SEE EDITIONS':
-                logger.warning('game \"%s\" has different editions available, this is not yet supported', name)
+                logger.debug('processing editions for \"%s\"', name)
+                # used to know the number of elements and titles
+                # without having to query on every refresh
+                editions_addons_titles = WebDriverWait(browser, TIMEOUT).until(
+                    EC.visibility_of_all_elements_located((
+                        By.XPATH, "//div[contains(@class,'Editions-title') or contains(@class, 'AddOns-title')]"
+                    )))
+                for x in range(len(editions_addons_titles)):
+                    editions_addons_titles[x] = editions_addons_titles[x].text
+
+                for t in range(len(editions_addons_titles)):
+                    editions_addons_buttons = WebDriverWait(browser, TIMEOUT).until(
+                        EC.visibility_of_all_elements_located((
+                            By.XPATH,
+                            "//div[contains(@class,'Editions') or contains(@class, 'AddOns')]//div[contains(@class,'PurchaseButton-ctaButtons')]//button"
+                        )))
+
+                    if editions_addons_buttons[t].text == 'OWNED':
+                        logger.info('\"%s\" already owned', editions_addons_titles[t])
```

Comment:
Agree with @mattjmorgan . I think it should log not just edition name but game name too (if you're not debugging game name wouldn't get in log otherwise):

```python
logger.info('\"%s: %s\" already owned', name, editions_addons_titles[t])
```

Reply:
Added the name of the game to the log message as '<game name> - <edition or addon name>'

Reply:
With the current game we have an example of an edition that has the same exact edition name as the game itself so it will show as `"Faeria - Faeria" already owned`

Reply:
That doesn't bother me 😜